### PR TITLE
Use release/1 build of dpl-react

### PR DESCRIPTION
**Pull request by spaceo**
#### Link to issue

https://reload.atlassian.net/browse/DDFSOEG-98

#### Description

This PR:
* Enables the dpl_react module so the react apps can be rendered
* Imports the latests heads of the release/1 branch of dpl-react and dpl-design-system
* Adds the search header library
* Implements the search_header library in the header of the Novel theme.
#### Screenshot of the result

image
#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

This PR is using deliberately using the release/1 assets of dpl-design-system and dpl-react. The should of course be corrected to latest main in the future.

### CODE REVIEW

File: web/themes/custom/novel/novel.theme [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ *   Theme variables.
+ */
+function novel_preprocess_page(array &$variables): void {
+  $data = [
+    'search-header-url' => '/search',
+    'alt-text' => t('Search field'),
+    'input-placeholder' => t('Start typing in order to search'),
+  ];
+
+  $variables['search_header'] = dpl_react_render('search-header', $data);
```
Comment (kasperg) [inline web/themes/custom/novel/novel.theme:21]: This creates a dependency between Novel and DPL React. Please remember to declare it in `novel.info.yml`.

File: web/themes/custom/novel/novel.theme [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * Implements template_preprocess_page().
+ *
+ * @param mixed[] $variables
+ *   Theme variables.
+ */
+function novel_preprocess_page(array &$variables): void {
+  $data = [
+    'search-header-url' => '/search',
+    'alt-text' => t('Search field'),
+    'input-placeholder' => t('Start typing in order to search'),
```
Comment (kasperg) [inline web/themes/custom/novel/novel.theme:18]: Please add a context. This is required [according to our coding guidelines](https://github.com/danskernesdigitalebibliotek/dpl-cms/blob/main/documentation/code-guidelines.md#translations).

We should consider requesting some form of automation alerting developers of this. I think many people will forget this.

File: web/themes/custom/novel/novel.theme [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * Implements template_preprocess_page().
+ *
+ * @param mixed[] $variables
+ *   Theme variables.
+ */
+function novel_preprocess_page(array &$variables): void {
+  $data = [
+    'search-header-url' => '/search',
```
Comment (kasperg) [inline web/themes/custom/novel/novel.theme:16]: Consider adding a TODO explaining that this url should be changed once we have a proper search result page.

File: web/themes/custom/novel/templates/layout/footer.html.twig [hunk 4]
```diff
@@ -1,35 +1,35 @@
...
+              class="accordion__button px-48"
               aria-expanded="false"
               data-accordion-trigger="true"
             >
               Om Bibliotekerne
               <img
-                class="accordion-button--arrow"
-                src="/themes/custom/novel/assets/dpl-design-system/icons/basic/icon-triangle.svg"
+                class="accordion__button-arrow"
+                src="icons/basic/icon-triangle.svg"
```
Comment (kasperg) [inline web/themes/custom/novel/templates/layout/footer.html.twig:27]: Please fix the url to the icon. The problem with the 404 resides here.

File: web/themes/custom/novel/templates/layout/footer.html.twig [hunk 5]
```diff
@@ -121,25 +123,25 @@
...
-      <div class="footer-separator mt-24"></div>
-      <div class="footer-site-info mt-24">
-        <div class="footer-site-info--links">
+      <div class="footer__separator mt-24"></div>
+      <div class="footer__site-info mt-24">
+        <div class="footer__site-info-links">
           <ul>
             <li>
-              <a href="/" class="footer-site-info--link">
+              <a href="" class="footer__site-info-link">
```
Comment (kasperg) [inline web/themes/custom/novel/templates/layout/footer.html.twig:131]: Please fix the href.

File: web/themes/custom/novel/templates/layout/header.html.twig [hunk 6]
```diff
@@ -1,162 +1,150 @@
...
-          >
-            <div class="pagefold-triangle-small"></div>
-            <img src="/themes/custom/novel/assets/dpl-design-system/icons/basic/icon-menu.svg" alt="List of bookmarks" />
-          </div>
-          <div class="header-menu--navigation--logo">
-            <div class="logo-fallback-text">
-              <p class="logo-fallback-text--name">Lyngby-Taarbæk</p>
-              <p class="logo-fallback-text--libraries">Bibliotekerne</p>
+
+  <header class="header">
```
Comment (kasperg) [inline web/themes/custom/novel/templates/layout/header.html.twig:2]: Please look into indentation and whitespace in this file.

All the content seems to be indented and there is a blank line at the start of the file.

File: composer.json [hunk 7]
```diff
@@ -20,7 +20,7 @@
                 "version": "0.1.0",
                 "type": "drupal-library",
                 "dist": {
-                    "url": "https://github.com/danskernesdigitalebibliotek/dpl-react/releases/download/0.1.0/dist.zip",
+                    "url": "https://github.com/danskernesdigitalebibliotek/dpl-react/releases/download/release-release%2F1/dist.zip",
```
Comment (kasperg) [inline composer.json:23]: Consider if we should refer to a tag here. `release/1` seems to be a moving target.

Whatever we do please update `version` accordingly.

File: composer.json [hunk 8]
```diff
@@ -33,9 +33,9 @@
             "package": {
                 "name": "danskernesdigitalebibliotek/dpl-design-system",
                 "type": "drupal-library",
-                "version": "v1.0.2",
+                "version": "v1.0.3",
                 "dist": {
-                    "url": "https://github.com/danskernesdigitalebibliotek/dpl-design-system/releases/download/v1.0.2/dist.zip",
+                    "url": "https://github.com/danskernesdigitalebibliotek/dpl-design-system/releases/download/release-release%2F1/dist.zip",
```
Comment (kasperg) [inline composer.json:38]: Consider if we should refer to a tag here. `release/1` seems to be a moving target.

Whatever we do please update version accordingly.

File: web/themes/custom/novel/templates/layout/page.html.twig [hunk 9]
```diff
@@ -47,6 +47,7 @@
   {% include '@novel/layout/header.html.twig'
     with {
     'header': page.header,
+    'search_header': search_header
```
Comment (kasperg) [inline web/themes/custom/novel/templates/layout/page.html.twig:50]: Consider if there is a patttern regarding naming.

Other variables in the template seems to suggest a pattern of using `.` as a separator to scope elements.

**Review by kasperg — CHANGES_REQUESTED** [review]
:+1: I have a couple of things I would like you to look into before this is ready to merge.

File: web/themes/custom/novel/novel.theme (hunk 1 above)
Comment (spaceo) [reply web/themes/custom/novel/novel.theme:21]: 👍

File: web/themes/custom/novel/novel.theme (hunk 3 above)
Comment (spaceo) [reply web/themes/custom/novel/novel.theme:16]: 👍

File: web/themes/custom/novel/templates/layout/footer.html.twig (hunk 5 above)
Comment (spaceo) [reply web/themes/custom/novel/templates/layout/footer.html.twig]: 👍

File: web/themes/custom/novel/templates/layout/header.html.twig (hunk 6 above)
Comment (spaceo) [reply web/themes/custom/novel/templates/layout/header.html.twig:2]: 👍

File: composer.json (hunk 8 above)
Comment (spaceo) [reply composer.json:38]: 👍

File: web/themes/custom/novel/templates/layout/footer.html.twig (hunk 4 above)
Comment (spaceo) [reply web/themes/custom/novel/templates/layout/footer.html.twig]: 👍

File: web/themes/custom/novel/templates/layout/page.html.twig (hunk 9 above)
Comment (spaceo) [reply web/themes/custom/novel/templates/layout/page.html.twig]: 👍

File: web/themes/custom/novel/novel.theme (hunk 2 above)
Comment (spaceo) [reply web/themes/custom/novel/novel.theme]: 👍

File: web/themes/custom/novel/novel.theme [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+  $metatag = [
+    '#tag' => 'meta',
+    '#attributes' => [
+      'name' => 'description',
+      'content' => t('This is the meta description of the DPL CMS site'),
+    ],
+  ];
+
+  $page['#attached']['html_head'][] = [$metatag, 'description'];
+}
```
Comment (kasperg) [inline web/themes/custom/novel/novel.theme:43]: Please remember to remove this. It is currently not needed to improve pagespeed score and in the future we will probably want to address this using the `metatag` module. Also the current implementation does not seem to work 😅 .

**Review by kasperg — APPROVED** [review]
:+1: LGTM

**Review by turegjorup — APPROVED** [review]

**Pull request merged**